std_execution.bs

```diff
@@ -3994,6 +4003,161 @@ from the operation before the operation completes.
+
+3. [<i>Note:</i> Implementations are encouraged to use an intrusive queue of operation states to hold the work units to make scheduling allocation-free. -- <i>end note</i>]
+
+    <pre highlight="c++">
+    class run_loop {
+      // [execution.contexts.run_loop.types] Associated types
+      class <i>run-loop-scheduler</i>; // exposition only
+      class <i>run-loop-sender</i>; // exposition only
+      struct <i>run-loop-opstate-base</i> { // exposition only
+        virtual void execute() = 0;
+        void start() noexcept { loop_->push_back(this); }
+        run_loop* loop_;
+        <i>run-loop-opstate-base</i>* next_;
+      };
+      template&lt;receiver_of R>
+        using <i>run-loop-opstate</i> = <i>unspecified</i>; // exposition only
+
+      // [execution.contexts.run_loop.members] Member functions:
+      <i>run-loop-opstate-base</i>* pop_front() noexcept; // exposition only
+      void push_back(<i>run-loop-opstate-base</i>*) noexcept; // exposition only
+
+     public:
+      // [execution.contexts.run_loop.ctor] Constructor:
+      run_loop() noexcept;
+      run_loop(run_loop&&) = delete;
+
+      // [execution.contexts.run_loop.members] Member functions:
+      <i>run-loop-scheduler</i> get_scheduler();
+      void run();
+      void finish();
+    };
+    </pre>
+
+#### Associated types <b>[execution.contexts.run_loop.types]</b> #### {#spec-execution.contexts.run_loop.types}
+
+    <pre highlight="c++">
+    class <i>run-loop-scheduler</i>;
+    </pre>
+
+  1. <code><i>run-loop-scheduler</i></code> is an implementation defined type that models the `scheduler` concept.
+
+  2. Instances of <code><i>run-loop-scheduler</i></code> remain valid until the end of the lifetime of the `run_loop` instance from which they were obtained.
+
+  3. Two instances of <code><i>run-loop-scheduler</i></code> compare equal if and only if they were obtained from the same `run_loop` instance.
+
+  4. Let <code><i>sch</i></code> be an expression of type <code><i>run-loop-scheduler</i></code>. The expression  <code>execution::schedule(<i>sch</i>)</code> is not potentially throwing and has type <code><i>run-loop-sender</i></code>.
+
+  <pre highlight="c++">
+  class <i>run-loop-sender</i>;
+  </pre>
+
+  1. <code><i>run-loop-sender</i></code> is an implementation defined type that models the `sender_of` concept; <i>i.e.,</i> <code>sender_of&lt;<i>run-loop-sender</i>></code> is `true`. Additionally, the types reported by its `error_types` associated type is `exception_ptr`, and the value of its `sends_done` trait is `true`.
+
+  2. Instances of <code><i>run-loop-sender</i></code> remain valid even after the end of the lifetime of the <code><i>run-loop-scheduler</i></code> instance from which they were obtained.
```

Comment:
I don't think this has normative meaning, maybe turn this into a note?

Reply:
I've fixed this. The senders are valid until the end of the lifetime of their associated run_loop instances.